subsys/bluetooth/mesh/proxy_msg.c

```diff
@@ -78,6 +78,11 @@ ssize_t bt_mesh_proxy_msg_recv(struct bt_conn *conn,
 	const uint8_t *data = buf;
 	struct bt_mesh_proxy_role *role = &roles[bt_conn_index(conn)];
 
+	if (net_buf_simple_tailroom(&role->buf) < len - 1) {
```

Comment:
Please check if it is possible to ever receive a length of zero?

Reply:
Checking that the length is not zero happens in the functions those call `bt_mesh_proxy_msg_recv` as the preliminary checking before passing data for further handling.
Here: 
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/proxy_srv.c#L90-L93
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/pb_gatt_srv.c#L85-L88
https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/mesh/gatt_cli.c#L70-L73

I do not think it is worth to check here one more time if I understood your concern correctly.